src/main/scala/org/deepdive/ddlog/DeepDiveLogParser.scala

```diff
@@ -449,6 +450,11 @@ class DeepDiveLogParser extends JavaTokenParsers {
     } =>
       val weights = FactorWeight(rule.annotations find (_ named "weight") map (_ exprs) get)
       val valueExpr = (rule.annotations find (_ named "value") flatMap (_ expr))
+      val ruleName = (rule.annotations find (_ named "name") flatMap (_ expr) map {
+        case StringConst(x) => x
+        case _ => sys.error(s"Invalid rule name:\n${DeepDiveLogPrettyPrinter.print(rule)}")
+      })
```

Comment:
I guess we could fold this check into the new `ruleName` member too.  `val` is probably necessary as `def` won't throw the error at parse time.